File: test-suite/src/test/java/io/micronaut/docs/aws/sdk/v2/AwsClientFactoryTest.java
package io.micronaut.docs.aws.sdk.v2;

import io.micronaut.context.ApplicationContext;
import io.micronaut.context.annotation.Property;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;
import software.amazon.awssdk.services.rekognition.RekognitionAsyncClient;
import software.amazon.awssdk.services.rekognition.RekognitionClient;

import static org.junit.jupiter.api.Assertions.assertEquals;

@Property(name = "spec.name", value = "AwsClientFactoryTest")
@MicronautTest(startApplication = false)
class AwsClientFactoryTest {

    @Inject
    ApplicationContext applicationContext;

    @Test
    void itCanCreateSyncClients() {
        RekognitionClient client = applicationContext.getBean(RekognitionClient.class);

        assertEquals(RekognitionClient.SERVICE_NAME, client.serviceName());
    }

    @Test
    void itCanCreateAsyncClients() {
        RekognitionAsyncClient client = applicationContext.getBean(RekognitionAsyncClient.class);

        assertEquals(RekognitionClient.SERVICE_NAME, client.serviceName());
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/aws/sdk/v2/AwsClientFactoryTest.kt
package io.micronaut.docs.aws.sdk.v2

import io.micronaut.context.ApplicationContext
import io.micronaut.context.annotation.Property
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import software.amazon.awssdk.services.rekognition.RekognitionAsyncClient
import software.amazon.awssdk.services.rekognition.RekognitionClient

@Property(name = "spec.name", value = "AwsClientFactoryTest")
@MicronautTest(startApplication = false)
class AwsClientFactoryTest {

    @Inject
    lateinit var applicationContext: ApplicationContext

    @Test
    fun itCanCreateSyncClients() {
        val client = applicationContext.getBean(RekognitionClient::class.java)

        assertEquals(RekognitionClient.SERVICE_NAME, client.serviceName())
    }

    @Test
    fun itCanCreateAsyncClients() {
        val client = applicationContext.getBean(RekognitionAsyncClient::class.java)

        assertEquals(RekognitionClient.SERVICE_NAME, client.serviceName())
    }
}
